### ISSUE TITLE
opcodes do not correctly implement padding

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#1514]
If one writes
```rust
use bitcoin::blockdata::opcodes;

fn main() {
    let op = opcodes::all::OP_NOP;
    println!("{:>10}", 0);
    println!("{:>10}", op);
}
```
The expected output is:
```
         0
    OP_NOP
```
But the actual output is:
```
         0
OP_NOP
```

**Comment by tcharding** [comment]
Hint: The reason is because we use `write!` we loose the options to the formatter. Discovered while investigating https://github.com/rust-bitcoin/rust-miniscript/issues/323

**Comment by Kixunil** [comment]
Should we just use `.pad` and accept that the string may be truncated (to useless strings like `OP_`) or do we want to roll our own, non-truncating helper?

**Comment by apoelstra** [comment]
I vote in favor of truncating (even to the point of uselessness). This is the expected behavior when formatting strings, and will signal to the end user that they need to widen their terminal (or something).

**Comment by td-bn** [comment]
Might be a silly question but what would be wrong with something like this?
```rust
let prefex = "OP_";
...
format!("{}NOP", prefix).fmt(f)
```
I imagine this would format as expected by the user.

**Comment by Kixunil** [comment]
@td-bn yeah, it just allocates needlessly and is generally over-compliated. I'd just use a macro:

```rust
macro_rules! display_opcode {
    ($($opcode:ident),+ $(,)?) => {
       $(
            $opcode => fmt::Display::fmt(stringify!($opcode), f),
        )+
    }
}

match self {
    display_opcode! {
        OP_PUSHDATA1,
        OP_PUSHDATA2,
        // ...
        // skip doing "clever" things with OP_PUSHNUM_x, not worth it.
    }
}
```

**Comment by td-bn** [comment]
For combined, I was thinking of something like this, but it means we'd have to pass code and doc string to macro:
```rust
macro_rules! all_opcodes {
    ($($op:ident => $val:expr, $doc:expr),*) => {
         pub mod all {
             use super::All;
                 $(
                     /// $doc
                     pub const $op: All = All { code: $val};
                 )*
         }

         impl fmt::Display for All {
             fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
                  match *self {
                    $(
                         $op => fmt::Display::fmt(stringify!($op), f),
                     )+
                 }
             }
         }

     }
 }

```
If that's fine, I can make these changes and push again.

**Comment by Kixunil** [comment]
Yeah, that's fine! With a bit of fiddling I think you could make it accept `///` as doc comments (they are converted to attributes under the hood), but what you suggest is fine too. I'd just use `;` instead of `,` to separate the triples.

**Comment by tcharding** [comment]
Fixed by #1528, right?
